woocommerce/woocommerce-ios
# Shipping Labels: validate shipping label address screen

Part of #2973 

## Description
As part of Shipping Labels M2, we need to present the address validation form. In this PR, I implemented the screen that shows the basic fields and the confirmation button below. There is still no logic behind it and some UI customizations will be applied later. All the missing things will be handled in separate PRs. 


## Changes
- Now the field `address` in `ShippingLabelAddressVerification` model is declared as optional.
- Implemented the new `ShippingLabelAddressValidationViewController` which represents the validation screen for the address (origin and destination). It contains all the cell configuration for the address fields, the keyboard management, and the confirmation button. 
- Implemented the `ShippingLabelAddressValidationViewModel` which is useful for configuring `ShippingLabelAddressValidationViewController`. For the moment, it just returns the Sections without any logic.
- The `ShippingLabelFormViewController` is now able to present the `ShippingLabelAddressValidationViewController`.
- The models `ShippingLabelAddressVerification` and `ShippingLabelAddressValidationResponse` are now exposed throught Yosemite.
- Implemented a method in `ShippingLabelFormViewModel` that is useful for transforming an `Address` into a `ShippingLabelAddress`.

## Testing
1. Launch the app in debug mode (because of the feature flag).
2. Open an order detail.
3. Press the button "Create Shipping Label".
4. Press the button "Continue" in the Create Shipping Label form.
5. The new screen for validating an address should be presented.

## Screenshot
<img src="https://user-images.githubusercontent.com/495617/107958740-c24ad000-6fa2-11eb-9cac-00ace95eaa60.png" width=300 />


Update release notes:
